business_rules/engine.py

```diff
@@ -90,15 +144,25 @@ def _do_operator_comparison(operator_type, operator_name, comparison_value):
 
     def fallback(*args, **kwargs):
         raise AssertionError("Operator {0} does not exist for type {1}".format(
-            operator_name, operator_type.__class__.__name__))
+            operator_name, operator_type.__class__.__name__)
+        )
 
     method = getattr(operator_type, operator_name, fallback)
     if getattr(method, 'input_type', '') == FIELD_NO_INPUT:
         return method()
     return method(comparison_value)
 
 
-def do_actions(actions, defined_actions):
+def do_actions(actions, defined_actions, override_params=None):
+    """
+    Executes defined actions.
+    :param actions: actions to be exectuted
+    :param defined_actions: scope containing all relevant actions
+    :param override_params: actions' params overrides coming from variables
+    :return:
```

Comment:
Bonus points if you add type annotations

Reply:
wasn't sure on adding types to the functions' signatures, so I added type directives to the docstrings